noir/noir-repo/acvm-repo/acir/src/circuit/mod.rs

```diff
@@ -37,6 +39,7 @@ pub enum ExpressionWidth {
 #[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Default)]
 pub struct Program {
     pub functions: Vec<Circuit>,
+    pub unconstrained_functions: Vec<BrilligBytecode>,
```

Comment:
Is the plan for all unconstrained functions to live here? this is going to require a bunch of work for handling errors and debug metadata (OpcodeLocation) I think. But it's work that I guess is going to be needed for multiple ACIR functions too so I guess it's fine

Reply:
Yeah all unconstrained functions that are entry points will live here. As you said though, the work to handle errors and debug metadata is already necessary due to multiple ACIR functions as you stated. In fact, for Brillig calls it is much simpler as we during ACIR gen we can save the generated brilligs and appropriately insert the debug/location information into the ACIR debug info. In fact I already did some of this handling here https://github.com/AztecProtocol/aztec-packages/pull/5737.